app/controllers/timer_sessions_controller.rb

```diff
@@ -93,6 +93,12 @@ def update
     end
   end
 
+  def self.offset_for_time_zone(current_user)
+    return 0 unless current_user&.preference&.time_zone
+
+    Time.zone.now.in_time_zone(current_user.preference.time_zone).utc_offset / 1.hour
+  end
```

Comment:
Rather make this a private instance method and use `@current_user` in it. Less indirection.